### DISCUSSION TITLE
Is it possible to generate breast cancer data using some flag?

### DISCUSSION THREAD

**Discussion by nehaghub** [synthetichealth/synthea#1200] [General]
Hi Team,

I am new to this synthea data generator and i am not able to generate breast cancer patient data.

Kindly help me in finding the correct command to generate breast cancer patient data.

That will be very helpful to me.

Thanks
Neha

**Comment by jawalonoski** [comment]
See https://github.com/synthetichealth/synthea/wiki/Keep-Patients-Module

Synthea is not designed to create only breast cancer data (or patients with only one single ailment). It is designed to create a **complete** longitudinal health record: all diagnoses, all treatments, for the patients entire life.

You can use the Keep Patients Module to filter patients according to criteria you define (e.g., must have breast cancer). If you want to filter out data beyond that (e.g., all data not relevant to cancer treatment), you'll have to further post-process the results (which is beyond the scope of this project).

**Reply by nehaghub** [reply]
Hi,
Thanks for your quick reply. But, I did not find a single record of any cancer patient when i am running this project.
If i am not wrong according to below link , it is generating data for only the diseases for which java module is written.
https://github.com/synthetichealth/synthea/wiki/Adding-a-Module



[Lifecycle](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/LifecycleModule.java) - birth, death, height, weight, and other vital signs
[Encounters](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/EncounterModule.java) - medical encounters are performed, diagnoses are made, and the results are  #recorded
[Cardiovascular Disease](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/CardiovascularDiseaseModule.java) - coronary heart disease, myocardial infarction, cardiac arrest, stroke, atrial fibrillation
[Immunizations](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/Immunizations.java) - vaccines and immunization schedules

I think i have to write Breast cancer java module. Please correct me if i am wrong.

Thanks
Neha

**Reply by jawalonoski** [reply]
**No, do not write a Java module.**

From the top of the page you linked:

> We strongly encourage new modules be developed using the [Generic Module Framework](https://github.com/synthetichealth/synthea/wiki/Generic-Module-Framework). Java modules should be used only when the GMF does not support your specific use case.

It turns out that the GMF modules **do** support your use-case: https://synthetichealth.github.io/module-builder/#breast_cancer

Please generate at least 1000 patients, preferably 10000, and use a **Keep Patients Module** that requires a diagnosis of Breast Cancer, `254837009`, `Malignant neoplasm of breast (disorder)`.

Also, change the `exporter.years_of_history = 0` in your `synthea.properties` file so you are not filtering out history.

**Reply by nehaghub** [reply]
Hi,
Thank you for your help and support.
I have executed run_synthea script with the option **keep patients module**.
below is my command
**./run_synthea -p 1000 -k src/main/resources/modules/breast_cancer.json**

But no patients are generated with above command. Below message is coming
```
Waiting for threads to finish... java.util.concurrent.ThreadPoolExecutor@58658f63[Shutting down, pool size = 8, active threads = 8, queued tasks = 992, completed tasks = 0]

java.lang.IllegalArgumentException: Vital signs must have finite values - NaN is invalid
        at org.mitre.synthea.world.agents.Person.setVitalSign(Person.java:467)
        at org.mitre.synthea.modules.WeightLossModule.manageYearTwoThroughFive(WeightLossModule.java:165)
        at org.mitre.synthea.modules.WeightLossModule.process(WeightLossModule.java:82)
        at org.mitre.synthea.engine.Generator.updatePerson(Generator.java:700)
        at org.mitre.synthea.engine.Generator.createPerson(Generator.java:656)
        at org.mitre.synthea.engine.Generator.generatePerson(Generator.java:470)
        at org.mitre.synthea.engine.Generator.lambda$run$3(Generator.java:379)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Records: total=0, alive=0, dead=0
RNG=47
Clinician RNG=5644

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3.3/userguide/command_line_interface.html#sec:command_line_warnings
```

**Even i have tried to generate only one patient with keep patient module using below command after updating below configuration in synthea.properties file.**

**generate.max_attempts_to_keep_patient = 100000  [updated from 1000 to 100000]**
**exporter.years_of_history = 0**




**./run_synthea -p 1 -k src/main/resources/modules/breast_cancer.json**

Scripts run for 2 hours but still no patient generated.

Please let me know if I am missing something.

Thanks
Neha

**Reply by jawalonoski** [reply]
Please read https://github.com/synthetichealth/synthea/wiki/Keep-Patients-Module

You have to create a new keep module.

The breast cancer module `src/main/resources/modules/breast_cancer.json` is **not** your keep module. It is the disease module that creates breast cancer patients.

**Comment by nehaghub** [comment]
Hi, 
I am not able to understand. I wrote below module to keep breast cancer patient
`{
  "name": "keep",
  "remarks": [
    "A blank module"
  ],
  "states": {
    "Initial": {
      "type": "Initial",
      "conditional_transition": [
        {
          "transition": "Keep",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "breast_cancer_survival",
            "operator": "==",
            "value": "no"
          }
        },
        {
          "transition": "Terminal"
        }
      ]
    },
    "Terminal": {
      "type": "Terminal"
    },
    "Keep": {
      "type": "Terminal"
    }
  },
  "gmf_version": 2
}`


Patients are coming from breast cancer harmone_diagnosis.json file.

But, **I want patient from below module which includes absence of left breast or right breast patient.**
**Please guide me** if you can. Below is my disease module file that I have modified for my use case.
[breast_cancer.docx](https://github.com/synthetichealth/synthea/files/10064477/breast_cancer.docx)

I have wrote in word file for attachment because json attachment not supported.

Thanks
Neha

**Comment by lhs-open** [comment]
Neha, 
In addition to Jay's approach, you may try a different approach to get patients with lung cancers: 
1. run synthea to create a patient population using all default modules. The population will include patients with lung cancer by default.
2. select lung cancer patients in the resulting patients.csv file with the standard snomed-ct codes you need.
